### PR TITLE
Add keywords endpoint and refactor index into endpoint partials

### PR DESCRIPTION
**Pull request by jurgenwerk**

### CODE REVIEW

File: source/endpoints/groups.md.erb [hunk 1]
```diff
@@ -116,7 +116,7 @@ A group is a top-level container for your URLs and global views.
 ## List groups
 
 ```shell
-  curl 'https://api.nightwatch.io/api/v1/url_groups?access_token=ACCESS_TOKEN' \
+  curl 'https://api.nightwatch.io/api/v1/url_groups?access_token=ACCESS_TOKEN' \\
```
Comment (szlend) [inline source/endpoints/groups.md.erb]: Why double backslashes?

File: source/endpoints/access_token.md.erb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Access Token
+
+## Obtain an access token for your user
+
+```shell
+curl 'https://api.nightwatch.io/api/v1/token' \
```
Comment (szlend) [inline source/endpoints/access_token.md.erb:6]: Single backslash here

File: source/endpoints/keywords.md.erb [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+sort      | The name of the field to sort by. <br><br>Possible values: id, query, google_gl, google_hl, last_processed_at, created_at, updated_at, position, position_type, position_organic, position_places, position_places_image, position_changed_by, position_changed_status, position_changed_at, position_local_pack, position_featured_snippet, position_knowledge_panel, local_search, position_changed_during_last_day, position_changed_during_last_week, position_changed_during_last_month, mobile, engine, previous_position, best_position, available_serp_features, adwords_global_search_volume, adwords_global_average_cpc, adwords_local_search_volume, adwords_local_average_cpc.
+direction | The direction to sort by. <br><br>Possible values: asc, desc.
+search    | A general search string to filter by keywords' tags, result URL, location.
+
+## Add keywords
+
+```shell
+  curl 'https://api.nightwatch.io/api/v1/urls/URL_ID/keywords/batch_create?access_token=ACCESS_TOKEN' \\
+    -H 'Content-Type: application/json' \\
+    --data-binary '{"keywords":"keyword1\nkeyword2\nkeyword3","tags":[],"google_gl":"us","google_hl":"en","mobile":false,"desktop":true,"search_engine":"google"}' \\
```
Comment (szlend) [inline source/endpoints/keywords.md.erb]: Would be easier to read and maintain in multiple lines.

File: source/endpoints/keywords.md.erb [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+### Parameters
+
+Parameter | Description
+--------- | -----------
+keywords*  | Search queries separated by new lines (\n)
+google_hl* | Search language code (see <a href="https://developers.google.com/custom-search/docs/ref_languages">here</a> for available options)
+google_gl*    | Search geolocation code (see <a href="https://developers.google.com/custom-search/docs/xml_results_appendices#countryCodes">here</a> for available options)
+search_engine* | Search engine ('google', 'youtube', 'places', 'bing' or 'yahoo')
+adwords_location_id | Search location Criteria ID (see <a href="https://developers.google.com/ad-exchange/rtb/geotargeting">here</a> for available options)
+mobile | Search adjusted for mobile devices (true or false, default: true)
```
Comment (szlend) [inline source/endpoints/keywords.md.erb]: Is `true` really the default?

File: source/endpoints/keywords.md.erb [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+### HTTP Request
+
+`PUT https://api.nightwatch.io/api/v1/urls/URL_ID/keywords/KEYWORD_ID?access_token=ACCESS_TOKEN`
+
+### Parameters
+
+Parameter | Description
+--------- | -----------
+keyword[tags]  | Tags
```
Comment (szlend) [inline source/endpoints/keywords.md.erb]: Inconsistent. Parameters are always nested into a key (e.g. keyword). Why is is only mentioned in this example? Should be always or never.

File: source/endpoints/keywords.md.erb (hunk 3 above)
Comment (szlend) [reply source/endpoints/keywords.md.erb]: We should be consistent when we use or don't use `-X POST`.

File: source/endpoints/keywords.md.erb (hunk 5 above)
Comment (szlend) [reply source/endpoints/keywords.md.erb]: It should be more descriptive, like "An array of tags".

**Review by szlend — CHANGES_REQUESTED** [review]

File: source/endpoints/groups.md.erb (hunk 1 above)
Comment (jurgenwerk) [reply source/endpoints/groups.md.erb]: When using only one backslash, the rendition gets broken, like this:

image

But I discovered in some cases it also works with a single backslash. With double backslashes it works alway. So I opted into using double backslashes everywhere.

File: source/endpoints/keywords.md.erb (hunk 3 above)
Comment (jurgenwerk) [reply source/endpoints/keywords.md.erb]: Fixed.

File: source/endpoints/keywords.md.erb (hunk 4 above)
Comment (jurgenwerk) [reply source/endpoints/keywords.md.erb]: Of course not. Good catch. I carelessly copied this from RankTrackr's docs. Which are wrong. Added a ticket to fix it there too.

File: source/endpoints/keywords.md.erb (hunk 5 above)
Comment (jurgenwerk) [reply source/endpoints/keywords.md.erb]: Yes. Fixed.

**Comment by jurgenwerk** [comment]
@szlend bump

**Review by szlend — APPROVED** [review]

**Pull request merged**